Genetics of Rheumatoid Arthritis: 2019 Status
==========================================================

* [47](48-SNPs-Genetic-Risk-Score-EUR.txt)    SNPs in Genetic Risk Score in European (non-HLA 48 + HLA)
* 69    PADI,PTPN Variants in RA Risk
* 128   [GWAS-Meta Validated SNPs](GWAS-Meta-128-SNPs.20190208.vcf)
* [241](miRNASeedSNP.hg19.bed)   SNPs in the 3-UTR regions of VIP genes (miRNA-UTR3 double pairs)
* 334   [VIP functional variants](gnomad.exomes.r2.1.sites.rec.VIP.hg19.vcf.bed)
* 327   [eQTL for VIP genes](VIP.eQTL.327.SNPs.txt)
* 478   SNPs located in miRNAs seed region
* 907   [HLA functional SNPs](gnomad.exomes.r2.1.sites.rec.HLA.hg19.vcf.bed) (6:29671116-33106890)
* [1062](1062.gnomad.exomes.r2.1.sites.rec.eQTL.missense.set2.hg19.vcf.bed) eQTL + mis-sense SNPs with MAF>0.1% in East Asian
* 1375  [Cytoband (>10 records)](../cytoband/1375.gnomad.exomes.r2.1.sites.rec.RA-GWAS-Cytoband.hg19.vcf.bed) from GWAS-Catalog-RA records
* 1763  [Epigene Functional Variants](gnomad.exomes.r2.1.sites.rec.EpiVar.txt)
* 3325  [GWAS Immnue System Desases](GWAS-immnue-3325_SNP.hg19.bed)
* 3591  [40K regions of GWAS Immnue System Desases](gnomad.exomes.r2.1.sites.rec.40KGWASAID.merge.vcf.bed)
* [4678](https://raw.githubusercontent.com/Shicheng-Guo/HowtoBook/master/miRNAome/4678.UTR3miRNAsNP.EAS.MAF0.01.hg19.bed) SNPs in UTR3 of 2133 immnue genes
* 2518  [Functional SNPs located in 824 Immune-related genes](gnomad.exomes.r2.1.sites.rec.InnateDB.merge.vcf.bed)
* [1635](gnomad.genomes.r2.1.sites.rec.innateDbUTR3.merge.vcf.bed) miRNA-SNPs located in 824 Immune-related genes UTR3 regions ([raw 11842](InnateDB.UTR3.snp.txt) )
* [21327](gnomad.genomes.r2.1.sites.rec.GWASCatalog.ASA.merge.vcf.hg19.bed) GWAS-Catalog (68193 (GWAS-Catalog) -> 42630 (MAF>1%) -> 21327(-ASA))
* [4728](gnomad.exomes.r2.1.4728.rec.ReactomePathWay.immnueGene.hg19.vcf.bed)  Reactome database related immnue system genes (9 pathway and 1571 genes)
* [4344](4344.GWAS-Meta2-Meta3.merge.rsList) Download GWAS-Meta Summary Dataset and merge meta2 and meta3 since Meta1 have >10,000 SNPs
* [5819](gnomad.genomes.eQTL.TFBS.uni.hg19.bed) eQTL-TFBS Overlap Variants
* [7087](../eQTL/gnomad.genomes.eQTL.cpgSNP.uni.ASA.hg19.bed) CpGSNP-eQTL-Overlap Variants
* [12380](12380.ASA.gnomad.exomes.r2.1.sites.rec.refGene.hg19.vcf.bed) Genome-wide Functional SNPs (MAF>1%, mis,frame,gain,loss,splice)
* [14873](gnomad.exomes.r2.1.sites.rec.TotalCandidateGene.hg19.vcf.bed) MAF=0.1% Frame+Loss+Missense+Stop+Gain in [interesting Genes](TotalCandidateGene.hg19.bed)
* 6200  TFBS-SNPs in lncRNA promoter region
* 44326 CpGI-TFSB-DNase-BUR SNPs(CpGI.TFBS.DNase.BUR.hg19.bed)
* [2665](2665-GWAS-LD0.8-ASN.hg19.rsList.txt)  GWAS RA signficant; 789 SNPs; LD R2>0.8 SNPs (not include intron,HaploReg)
=========================================================================

[ASA-primary-SNPs](ASA.hg19.bed) 738,981 primary SNPs in ASA-CHINA ARRAY
* [12380](12380.ASA.gnomad.exomes.r2.1.sites.rec.refGene.hg19.vcf.bed) Genome-wide Functional SNPs (MAF>1%, mis,frame,gain,loss,splice) added to ASA-CHINA-GuangHua
* [15446](../eQTL/15446.MRCI.ASA.eQTL.hg19.MAF0.001.hg19.bed) eQTL-SNPs with MAF>0.1% + R2<0.9 in East Asian added to ASA-CHINA-GuangHua
* 10648 in [14873](gnomad.exomes.r2.1.sites.rec.TotalCandidateGene.hg19.vcf.bed) added into ASA-CHINA-GuangHua
* [4678](https://raw.githubusercontent.com/Shicheng-Guo/HowtoBook/master/miRNAome/4678.UTR3miRNAsNP.EAS.MAF0.01.hg19.bed) SNPs in UTR3 of 2133 immnue genes
* 1708 in [4728](gnomad.exomes.r2.1.4728.rec.ReactomePathWay.immnueGene.hg19.vcf.bed) added into ASA-CHINA-GuangHua
* 1257 in [8501](gnomad.exomes.r2.1.8501.rec.GHRA_ASA.hg19.vcf.bed) added into ASA-CHINA-GuangHua
* 35 in [128](GWAS-Meta-128-SNPs.20190208.vcf.bed) added into ASA-CHINA-GuangHua
* 0 in [48](48-SNPs-Genetic-Risk-Score-EUR.hg19.bed) added into ASA-CHINA-GuangHua
* 44 in [907](gnomad.exomes.r2.1.sites.rec.HLA.hg19.vcf.hg19.bed) added into ASA-CHINA-GuangHua
* 69 in [1037](gnomad.exomes.r2.1.sites.rec.HLA.hg19.vcf.bed) Full chr6:29671116-33106890 added into ASA-CHINA-GuangHua
* 552 in [1375](1375.gnomad.exomes.r2.1.sites.rec.RA-GWAS-Cytoband.hg19.vcf.bed) added into ASA-CHINA-GuangHua
* 1711 in [2665](2665-GWAS-LD0.8-ASN.hg19.rsList.hg19.bed) added into ASA-CHINA-GuangHua
* 1902 in [4301](GWAS-immnue-3325_SNP.hg19.bed) added into ASA-CHINA-GuangHua
* 1152 in [3591](gnomad.exomes.r2.1.sites.rec.40KGWASAID.merge.hg19.bed) added into ASA-CHINA-GuangHua
* 46 in [2518](gnomad.exomes.r2.1.sites.rec.InnateDB.merge.hg19.bed) added into ASA-CHINA-GuangHua
* 1361 in [1635](gnomad.genomes.r2.1.sites.rec.innateDbUTR3.merge.hg19.bed) added into ASA-CHINA-GuangHua
* 5091 in [57543](gnomad.genomes.eQTL.TFBS.uni.hg19.bed) added into ASA-CHINA-GuangHua
* 1494 in [5839](cpgSNPisland.AID.GWAS.SNP.hg19.bed) added into ASA-CHINA-GuangHua
* 324 in [325](T325.hg19.bed)  added into ASA-CHINA-GuangHua
* 361 in [632](hsa-miRNALD.hg19.bed) added into ASA-CHINA-GuangHua
* 60 in [115](HLA-TagSnp.hg19.bed) added into ASA-CHINA-GuangHua
* 10089 in [11818](UTR3-miRNA.hg19.bed) added into ASA-CHINA-GuangHua
=============================================================================

Finally, apply 1000 Genome CHB+CHS dataset to remove R2>0.8 SNPs (only keep 1 of them)
* [52895](MRCI.ASA.hg19.bed) in [69987](MRCI.hg19.bed) were selected to be customed probes 
